pkg/kv/kvserver/rangefeed/budget.go

```diff
@@ -16,11 +16,26 @@ import (
 	"sync/atomic"
 	"time"
 
+	"github.com/cockroachdb/cockroach/pkg/roachpb"
+	"github.com/cockroachdb/cockroach/pkg/rpc"
+	"github.com/cockroachdb/cockroach/pkg/util/envutil"
 	"github.com/cockroachdb/cockroach/pkg/util/mon"
 	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/errors"
 )
 
+// useBudgets controls if RangeFeed memory budgets are enabled. Overridable by
+// environment variable.
+var useBudgets = envutil.EnvOrDefaultBool("COCKROACH_USE_RANGEFEED_MEM_BUDGETS", true)
+
+// maxFeedFraction is maximum percentage of feed memory pool that could be
+// allocated to a single feed budget. Overridable by environment variable.
+var maxFeedFraction = envutil.EnvOrDefaultFloat64("COCKROACH_RANGEFEED_MEM_FRACTION", 0.1)
```

Comment:
This should have a justification. Have you done any testing to verify this with realistic workloads?

I suppose this would be on a per-range basis. We currently allow rangefeeds to use the entire SQL budget, right? If we assume a 32 GB node with `--max-sql-memory=0.25`, this would give 800 MB per range. That's more than the entire range size (512 MB), so maybe that's a bit excessive? If we cut rangefeeds to about 50% of the SQL budget then that's about 400 MB per range, which seems more reasonable, but of course that'd drop to 200 MB or 100 MB on smaller machines.

I suppose it's better to be lenient here, at least to begin with, so maybe this is fine. But it's worth verifying this during stability.

Reply:
If we set the rangefeed budget to 30% of SQL, this would give a 60 MB limit on an 8 GB machine. Maybe that's a bit small? Does it make sense for this to be relative to the SQL budget? Maybe it would be better to use a fixed limit, or something relative to the range size? Or maybe a minimum/maximum bound?

Reply:
(to be clear, I'm fine with punting any thorough testing and tuning here for stability and get it merged first -- but I think we need to do it before 22.1 goes out)